src/main/java/org/jooq/lambda/concurrent/Async.java

```diff
@@ -0,0 +1,20 @@
+package org.jooq.lambda.concurrent;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+
+public final class Async {
+    private Async() {
+
+    }
+
+    public static <U> CompletionStage<U> supplyAsync(Supplier<U> supplier, Executor executor) {
+        return SameExecutorCompletionStage.of(CompletableFuture.supplyAsync(supplier, executor), executor);
+    }
+
+    public static CompletionStage<Void> runAsync(Runnable runnable, Executor executor) {
+        return SameExecutorCompletionStage.of(CompletableFuture.runAsync(runnable, executor), executor);
+    }
```

Comment:
... in particular, that SPI logic might change in future JDK versions, and I'd like such change to apply automatically to `Async`